integrations/fallback_chat/src/haystack_integrations/components/generators/fallback_chat/chat_generator.py

```diff
@@ -0,0 +1,404 @@
+                       If individual generators have their own timeout settings, the effective
+                       timeout will be the minimum of this value and the generator's timeout.
+        """
+        if not generators:
+            msg = "'generators' must be a non-empty list"
+            raise ValueError(msg)
+        if timeout is None or timeout <= 0:
+            msg = "'timeout' must be a positive number of seconds"
+            raise ValueError(msg)
+
+        # Validation via duck-typing: require a callable 'run' method
+        for gen in generators:
+            if not hasattr(gen, "run") or not callable(gen.run):
+                msg = "All items in 'generators' must expose a callable 'run' method (duck-typed ChatGenerator)"
+                raise TypeError(msg)
+
+        self.generators = list(generators)
+        self.timeout = float(timeout)
+
+    # ---------------------- Serialization ----------------------
+    def to_dict(self) -> dict[str, Any]:
+        return default_to_dict(
+            self,
+            generators=[gen.to_dict() for gen in self.generators if hasattr(gen, "to_dict")],
+            timeout=self.timeout,
+        )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> FallbackChatGenerator:
+        # Reconstruct nested generators from their serialized dicts
+        init_params = data.get("init_parameters", {})
+        serialized = init_params.get("generators") or []
+        deserialized: list[Any] = []
+        for g in serialized:
+            # Use the generic component deserializer available in Haystack
+            holder = {"component": g}
+            deserialize_component_inplace(holder, key="component")
+            deserialized.append(holder["component"])  # type: ignore[assignment]
+        init_params["generators"] = deserialized
+        data["init_parameters"] = init_params
+        return default_from_dict(cls, data)
+
+    # ---------------------- Execution helpers ----------------------
+    def _get_effective_timeout(self, gen: Any) -> float:
+        """
+        Get the effective timeout for a generator.
+        Uses the minimum of FallbackChatGenerator timeout and the generator's own timeout.
+
+        :param gen: The generator to get timeout for
+        :return: The effective timeout in seconds
+        """
+        # Get the generator's timeout if it has one
+        gen_timeout = getattr(gen, "timeout", None)
+
+        # If generator has no timeout or it's None, use FallbackChatGenerator timeout
+        if gen_timeout is None:
+            return self.timeout
+
+        # Use the minimum of both timeouts
+        return min(self.timeout, float(gen_timeout))
+
+    def _run_single_sync(
+        self,
+        gen: Any,
+        messages: list[ChatMessage],
+        generation_kwargs: dict[str, Any] | None,
+        tools: (list[Tool] | Toolset) | None,
+        streaming_callback: StreamingCallbackT | None,
+    ) -> dict[str, Any]:
```

Comment:
@vblagoje I'm pretty confused as to what this is function is doing. Why are we running the calls to the generator within a ThreadPoolExecutor with a single worker?